### PR TITLE
Features/csvdownload

### PR DESCRIPTION
**Pull request by kennethedwards-bixal**
How to test: 

- Pull down
- `npm install`
-  `node buildcsvdata.js`

### CODE REVIEW

**Comment by scottqueen-bixal** [comment]
please include  a description of what the changes in the code accomplish at a high-level as well as link back to the ticket in JIRA that it closes

File: test/testjson.json [hunk 1]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (scottqueen-bixal) [inline test/testjson.json:1]: if this file and directory is not needed please remove

File: buildcsvdata.js [hunk 2]
```diff
@@ -1,124 +1,162 @@
...
-          } else {
-            console.log('error', error)
-          }
-        });
+    .catch((error) => {
+      console.log('ERROR IN DELETING FILE ---> ', error)
+    })
+}
+
+// eslint-disable-next-line consistent-return
```
Comment (scottqueen-bixal) [inline buildcsvdata.js:80]: if we are not linting at this point, we don't need to include these disabled comments

File: buildcsvdata.js [hunk 3]
```diff
@@ -1,124 +1,162 @@
...
-    //  * and then the commit message, and all data must be stringfied
-    let jsondata = JSON.stringify({
-        "message": "uploading csv file",
-        "content": buffered,
-        "sha" : shaValue,
-        "branch" : CIRCLE_BRANCH
+  axios(config)
+    .then(() => {
+      console.log('DONE DELETING')
+      // eslint-disable-next-line no-use-before-define
```
Comment (scottqueen-bixal) [inline buildcsvdata.js:72]: if we are not linting at this point, we don't need to include these disabled comments

File: buildcsvdata.js [hunk 4]
```diff
@@ -1,124 +1,162 @@
 if (process.env.NODE_ENV !== 'production') {
-  require('dotenv').config();
+  // eslint-disable-next-line global-require
```
Comment (scottqueen-bixal) [inline buildcsvdata.js:2]: remove if we ar not linting

**Comment by scottqueen-bixal** [comment]
if we are now naming the file `waivers.csv` then we should remove `waiverscsv.csv` from code base

**Comment by scottqueen-bixal** [comment]
Is this the expected output for piids?

Screen Shot 2022-02-07 at 1 34 07 PM

**Comment by scottqueen-bixal** [comment]
Is this the expected output for country data.countriesOfOriginAndUSContent?
Screen Shot 2022-02-07 at 1 35 19 PM

I'm not clear on what the client is expecting out of this data? for example, are they expecting header fields like `data.countriesOfOriginAndUSContent` or are they wanting it to be more human-readable?

I can see that we have asked the question in the ticket description https://cm-jira.usa.gov/browse/MIA-5, but never got answered?

**Comment by kennethedwards-bixal** [comment]
Yes, it's an empty array right now.
When Steph and I tested it in the new portal-forms-test, we added some
countries to see what the format will look like.
Right now, there are no waivers that have this entry filled in.

-Ken

**Comment by kennethedwards-bixal** [comment]
> Is this the expected output for piids?
> 
> Screen Shot 2022-02-07 at 1 34 07 PM

Yes, this is what is found in the json

File: buildcsvdata.js (hunk 4 above)
Comment (kennethedwards-bixal) [reply buildcsvdata.js:2]: Yea, I just cut and copied from devken to this new branch.

**Comment by kennethedwards-bixal** [comment]
PLEASE make sure your local .env file has `CIRCLE_BRANCH=features/csvdownload` for local testing

- When on your local computer, if you run `node buildcsvdata.js`, the script will run and push up a fresh copy of of waiverscsv.csv.  

- The proper format of the data will look like it is in table format. 

This ticket attempts to close MIA-222 and MIA-159 
https://cm-jira.usa.gov/secure/RapidBoard.jspa?rapidView=3372&projectKey=MIA&view=detail&selectedIssue=MIA-222

File: buildcsvdata.js (hunk 2 above)
Comment (kennethedwards-bixal) [reply buildcsvdata.js:80]: done.

File: buildcsvdata.js (hunk 3 above)
Comment (kennethedwards-bixal) [reply buildcsvdata.js:72]: done

**Comment by clmedders** [comment]
I just walked through the testing steps and was able to get a CSV file so that is passing for me! 

To go off what @stephanie-graham said... is it possible to make it appear in multiple columns? I don't know if the raw data or how it is being uploaded to a CSV would allow what just thinking out loud!

**Comment by stephanie-graham** [comment]
@clmedders - the problem with multiple columns is that there is no limit (currently) on how many PIIDs or countries can be entered.  And we can't make infinite PIID and country columns.  And we can't estimate how many columns to make based on the data, unless we read all the data and calculate that before generating the CSV.  Do-able, but probably not worth it.

**Pull request merged**